### PR TITLE
feat(ci): enable cache in setup-python

### PR DESCRIPTION
**Pull request by 34j**
https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

This may speed up the test slightly. (although it will take longer to install poetry anyway)

### CODE REVIEW

**Comment by 34j** [comment]
This is useful in the rare case of a library that takes an inordinate amount of time to install. However I'm also not sure if it should be in the template. (How about commenting it out?)

I would appreciate it if you could take a look at other PRs.

File: project/.github/workflows/ci.yml [hunk 1]
```diff
@@ -48,11 +48,16 @@ jobs:
...
       - uses: actions/checkout@v4
+      - name: Install poetry
+        run: pipx install poetry
+      # https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages
       - name: Set up Python
         uses: actions/setup-python@v5
+        id: setup-python
         with:
           python-version: ${{ matrix.python-version }}
-      - uses: snok/install-poetry@v1.3.4
```
Comment (browniebroke) [inline project/.github/workflows/ci.yml:55]: I've just realised that this action uses the deprecated install script, and that the solution you suggest is the recommended one 👍

File: project/.github/workflows/ci.yml [hunk 2]
```diff
@@ -48,11 +48,16 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4
+      - name: Install poetry
+        run: pipx install poetry
+      # https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages
```
Comment (browniebroke) [inline project/.github/workflows/ci.yml:53]: I don't think we need that comment, it's pretty clear what this is doing 

```suggestion
```

**Comment by browniebroke** [comment]
Thanks for the effort and your patience!

**Pull request merged**